s3/session.go

```diff
@@ -47,13 +48,35 @@ func findBucketRegion(bucket string, config *aws.Config) (string, error) {
 }
 
 // TODO : unit tests
-func getAWSRegion(s3Bucket string, config *aws.Config, settings map[string]string) (string, error) {
+func GetAWSRegion(s3Bucket string, config *aws.Config, settings map[string]string) (string, error) {
 	if region, ok := settings[RegionSetting]; ok {
 		return region, nil
 	}
+
 	if config.Endpoint == nil ||
```

Comment:
This new extracting logic deserves extracting into a separate function, just like the `findBucketRegion`